go/host/db/batches.go

```diff
@@ -21,6 +23,9 @@ import (
 func (db *DB) GetHeadBatchHeader() (*common.BatchHeader, error) {
 	headBatchHash, err := db.readHeadBatchHash()
 	if err != nil {
+		if errors.Is(err, leveldb.ErrNotFound) { // todo: checking for impl specific error at this level feels like the abstractions are wrong
```

Comment:
The in-memory DB just returns empty data when it doesn't find things. But levelDB returns `ErrNotFound` so the behaviour in here has to cater for both...

I think ideally we should have in-mem DB return errutil.ErrNotFound and we should create a wrapper for level DB so it also returns errutil.ErrNotFound. Follow-up PR?

Reply:
yeah. sounds good